plugins/repository-hdfs/src/test/java/org/elasticsearch/repositories/hdfs/HdfsBlobStoreContainerTests.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.repositories.hdfs;
+
+import org.apache.hadoop.fs.FileContext;
+import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
+
+import java.io.IOException;
+
+public class HdfsBlobStoreContainerTests extends ESBlobStoreContainerTestCase {
+
+    @Override
+    protected BlobStore newBlobStore() throws IOException {
+        return new HdfsBlobStore(FileContext.getFileContext(), "", 100);
```

Comment:
I don't think this file context is initialized properly.  Look at `HdfsRepository` for how it is initialized, with the appropriate security checks.  You will need to create a temp dir for this, see the `createTempDir()` method that `ESBlobStoreContainerTestCase` inherits.  

Did you run this test?  What did it say, if anything?